Classes/ExpressionLanguage/DocumentTypeFunctionProvider.php

```diff
@@ -132,25 +134,35 @@ function($arguments, $cPid)
                     return $type;
                 }
 
+                // 3d object type if model parameter is not empty
+                if (!empty($queryParams['tx_dlf']['model'])) {
+                    return TYPE_3D_OBJECT;
+                }
+
                 // Load document with current plugin parameters.
                 $this->loadDocument($queryParams['tx_dlf'], $cPid);
                 if (!isset($this->document) || $this->document->getCurrentDocument() === null) {
                     return $type;
                 }
+
                 // Set PID for metadata definitions.
                 $this->document->getCurrentDocument()->cPid = $cPid;
 
                 $metadata = $this->document->getCurrentDocument()->getToplevelMetadata($cPid);
                 if (!empty($metadata['type'][0])) {
                     // Calendar plugin does not support IIIF (yet). Abort for all newspaper related types.
-                    if (
-                        $this->document->getCurrentDocument() instanceof IiifManifest
-                        && array_search($metadata['type'][0], ['newspaper', 'ephemera', 'year', 'issue']) !== false
+                    if (!($this->document->getCurrentDocument() instanceof IiifManifest
+                        && in_array($metadata['type'][0], ['newspaper', 'ephemera', 'year', 'issue']))
                     ) {
-                        return $type;
+                        $type = $metadata['type'][0];
                     }
-                    $type = $metadata['type'][0];
                 }
+
+                // set 3d object type if object element exist at the first element of LOGICAL struct map
+                if ($this->document->getCurrentDocument()->metadataArray['LOG_0001']['type'][0] == 'object') {
```

Comment:
We already have the toplevel metadata array at this point (see line 151).
```suggestion
                if ($metadata['type'][0] == 'object') {
```